exec/java-exec/src/main/java/org/apache/drill/exec/expr/ExpressionTreeMaterializer.java

```diff
@@ -267,6 +267,17 @@ public LogicalExpression visitBooleanOperator(BooleanOperator op, FunctionLookup
       return new BooleanOperator(op.getName(), args, op.getPosition());
     }
 
+    static public int computePrecision(LogicalExpression currentArg) {
+        int precision = currentArg.getMajorType().getPrecision();
```

Comment:
Well, if the type has no precision, what do you suggest?  Move the fix back to CastIntDecimal.java (calculate the precision based on the value), as I originally had it, but Jinfeng insisted was incorrect?
You two fellows, Jinfeng and Aman, can decide among yourselves.  Let me know when you're finished duking it out.  If there are any survivors, we can discuss on the hangout tomorrow. 

    On Monday, August 8, 2016 8:04 PM, Aman Sinha <notifications@github.com> wrote:
 

 In exec/java-exec/src/main/java/org/apache/drill/exec/expr/ExpressionTreeMaterializer.java:> @@ -267,6 +267,17 @@ public LogicalExpression visitBooleanOperator(BooleanOperator op, FunctionLookup
>        return new BooleanOperator(op.getName(), args, op.getPosition());
>      }
>  
> +    static public int computePrecision(LogicalExpression currentArg) {
> +        int precision = currentArg.getMajorType().getPrecision();
Did you consider checking whether the type has a precision or not using getMajorType().hasPrecision() ? That way, you would only call getPrecision() if it returned True and otherwise set the precision for INT, BIGINT. —
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.